src/main/java/com/friendly_machines/intellij/plugins/ideanative2debugger/impl/DebugProcess.java

```diff
@@ -74,20 +74,20 @@ public class DebugProcess extends XDebugProcess implements Disposable {
             new BreakpointHandler(this, BreakpointType.class),
     };
 
-    private GdbMiStateResponse gdbSend(String operation, String[] options, String[] parameters) {
+    private GdbMiStateResponse gdbSend(String operation, Collection<String> options, Collection<String> parameters) throws IOException, InterruptedException {
         return myMiFilter.gdbSend(operation, options, parameters);
     }
 
-    private Map<String, ?> gdbCall(String operation, String[] options, String[] parameters) throws GdbMiOperationException {
+    private Map<String, ?> gdbCall(String operation, Collection<String> options, Collection<String> parameters) throws GdbMiOperationException, IOException, InterruptedException {
```

Comment:
What do you think about adding an overload of gdbCall that leaves PARAMETERS off? I think it would make the vast majority of callers easier.

Reply:
I think I overdid it - where's the line between options and parameters (where only the first parameter element gets "--" prependen) ?

Reply:
>I think I overdid it - where's the line between options and parameters (where only the first parameter element gets "--" prepended) ?

No idea. On https://sourceware.org/gdb/onlinedocs/gdb/GDB_002fMI-Input-Syntax.html#GDB_002fMI-Input-Syntax there are options, then there's `--` and then there are parameters. If it's similar to a shell command line then the distinction is very minor.

I went through all the `gdbCall` calls we have and only dataDisassemble and dataDisassembleFile use `parameters` that are not empty. It was possible to test it with those. I tried leaving `mode` off: that's not allowed. I tried leaving `--` off and specifying `mode` regardless: that is allowed.